addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,335 @@
+                    pct_tax = self.env['account.tax']
+                down_payment_values.append([
+                    pct_tax,
+                    line['analytic_distribution'],
+                    line['quantity'] * fixed_tax.amount,
+                    account,
+                ])
+
+        downpayment_line_map = {}
+        for tax_id, analytic_distribution, price_subtotal, account in down_payment_values:
+            grouping_key = frozendict({
+                self._taxes_field_name(): tuple(sorted(tax_id.ids)),
+                'analytic_distribution': analytic_distribution,
+                'account_id': account,
+            })
+            downpayment_line_map.setdefault(grouping_key, {
+                **base_downpayment_lines_values,
+                self._taxes_field_name(): grouping_key[self._taxes_field_name()],
+                'analytic_distribution': grouping_key['analytic_distribution'],
+                order._qty_field_name(): 0.0,
+                'price_unit': 0.0,
+            })
+            downpayment_line_map[grouping_key]['price_unit'] += \
+                order.currency_id.round(price_subtotal * ratio)
+
+        return list(downpayment_line_map.values()), [key['account_id'] for key in downpayment_line_map]
+
+    def _prepare_base_downpayment_line_values(self, order):
+        self.ensure_one()
+        context = {'lang': order.partner_id.lang}
+        so_values = {
+            'name': _(
+                'Down Payment: %(date)s (Draft)', date=format_date(self.env, fields.Date.today())
+            ),
+            order._qty_field_name(): 0.0,
+            'order_id': order.id,
+            'discount': 0.0,
+            'is_downpayment': True,
+            'sequence': order.order_line and order.order_line[-1].sequence + 1 or 10,
+        }
+        del context
+        return so_values
+
+    def _prepare_invoice_values(self, order, dp_lines, accounts):
+        self.ensure_one()
+        return {
+            **order._prepare_invoice(),
+            'invoice_line_ids': [Command.create(
+                line._prepare_invoice_line(
+                    name=self._get_down_payment_description(order),
+                    quantity=1.0,
+                    **({'account_id': account.id} if account else {})
+                )
+            ) for line, account in zip(dp_lines, accounts)],
+        }
+
+    def _get_down_payment_description(self, order):
+        self.ensure_one()
+        context = {'lang': order.partner_id.lang}
+        if self.advance_payment_method == 'percentage':
+            name = _("Down payment of %s%%", self.amount)
+        else:
+            name = _("Down Payment")
+        del context
+        return name
+
+    def _consolidated_billing(self):
+        return False
+
+    def _account_type(self):
+        raise NotImplementedError # To override
+
+    def _account_category(self):
+        raise NotImplementedError # To override
+
+    def _taxes_field_name(self):
```

Comment:
Same here: give those functions docstrings. Also, your other mixins tend to start with the functions to override ; would be clearer if you did the same here